post_processing/pylbo/utilities/toolbox.py

```diff
@@ -225,3 +225,93 @@ def solve_cubic_exact(a, b, c, d):
+        Array of eigenfunction arrays of complex numbers.
+
+    Returns
+    -------
+    nzeroes : np.ndarray(dtype=int)
+        Counter array containing the number of zeroes of the real or imaginary part
+        of each input eigenfunction array.
+    """
+
+    nzeroes = np.array([], dtype=int)
+
+    for eigfunc in eigfuncs:
+        counter_real = 0
+        counter_imag = 0
+        sign_real_eigfunc = np.sign(np.real(eigfunc))
+        sign_imag_eigfunc = np.sign(np.imag(eigfunc))
+
+        for i in range(1, len(sign_real_eigfunc) - 1):
+            if sign_real_eigfunc[i - 1] * sign_real_eigfunc[i] == -1:
+                counter_real += 1
+            if sign_real_eigfunc[i - 1] * sign_real_eigfunc[i] == 0:
+                if sign_real_eigfunc[i - 2] * sign_real_eigfunc[i - 1] == 0:
+                    counter_real += 1
+
+            if sign_imag_eigfunc[i - 1] * sign_imag_eigfunc[i] == -1:
+                counter_imag += 1
+            if sign_imag_eigfunc[i - 1] * sign_imag_eigfunc[i] == 0:
+                if sign_imag_eigfunc[i - 2] * sign_imag_eigfunc[i - 1] == 0:
+                    counter_imag += 1
+
+        counter = min(counter_real, counter_imag)
+        nzeroes = np.append(nzeroes, counter)
+
+    return nzeroes
+
+
+def invert_continuum_array(cont, r_gauss, sigma):
+    """
+    Finds the location of resonance for eigenmode solutions having a real part that
+    might overlap with a continuum range.
+
+    Parameters
+    ----------
+    cont : numpy.ndarray
+        Array containing the range of a specific continuum. Automatically has the same
+        length as r_gauss, since it has the same shape as the equilibrium fields used
+        to calculate the continua. Can be complex, but only the resonance with the real
+        part is calculated.
+    r_gauss : numpy.ndarray
+        Array containing the grid on which equilibrium fields are defined.
+    sigma : complex
+        An eigenvalue solution of the generalized eigenvalue problem.
+
+    Returns
+    -------
+    r_inv : None, float
+        The location where there is resonance between the eigenmode and the continuum.
+        Returns None if there is no resonance with the specified continuum.
+    """
+
+    diff = np.sign(np.real(cont) - np.real(sigma))
+
+    if len(np.unique(diff)) < 2:
+        # There is no sign change, value is not contained in array.
+        return None
+    else:
+        for i in range(1, len(diff) - 1):
+            if diff[i] * diff[i - 1] < 0:
+                # Linear interpolation between the points where the sign change occurs.
+                r_inv = (np.real(sigma) - np.real(cont[i - 1])) / (
+                    np.real(cont[i]) - np.real(cont[i - 1])
+                ) * (r_gauss[i] - r_gauss[i - 1]) + r_gauss[i - 1]
+                return r_inv
+            elif diff[i] * diff[i - 1] == 0:
+                # The exact same value is in the continuum array, return it.
+                return r_gauss[i]
```

Comment:
Remove the `else` line and decrease indentation of the block by 1 level for readability, this `if-else` statement is automatically implied by `return None` (I like to try and avoid nested if-statements)